plugins/ocm-backend/src/helpers/config.ts

```diff
@@ -1,82 +1,84 @@
 import { Config } from '@backstage/config';
-import { Logger } from 'winston';
+import { OcmConfig } from '../types';
 
-const CLUSTERS_PATH = 'kubernetes.clusterLocatorMethods';
-const OCM_PREFIX = 'ocm';
-const CLUSTER_CONFIG_PATH = OCM_PREFIX;
-const HUB_CONFIG_PATH = `${OCM_PREFIX}.hub`;
+const KUBERNETES_PLUGIN_CONFIG = 'kubernetes.clusterLocatorMethods';
+const OCM_PREFIX = 'catalog.providers.ocm';
+const KUBERNETES_PLUGIN_KEY = 'kubernetesPluginRef';
 
-export const getConfigVariantPath = (config: Config): string => {
-  const hubConfig = config.has(HUB_CONFIG_PATH);
-
-  if (!hubConfig && !config.has(CLUSTER_CONFIG_PATH)) {
-    throw new Error(
-      `Neither hub or cluster configuration were specified at '${OCM_PREFIX}.' config`,
-    );
+export const deferToKubernetesPlugin = (config: Config): boolean => {
+  if (config.has(KUBERNETES_PLUGIN_KEY)) {
+    return true;
   }
-
-  return hubConfig ? HUB_CONFIG_PATH : CLUSTER_CONFIG_PATH;
+  return false;
 };
 
-export const getHubClusterName = (config: Config): string => {
-  const path = getConfigVariantPath(config);
-  const name =
-    config.getOptionalString(`${path}.name`) ||
-    config.getOptionalString(`${path}.cluster`);
-  if (!name) {
+export const getHubClusterFromKubernetesConfig = (
+  id: string,
+  config: Config,
+  globalConfig: Config,
+): Config => {
+  const name = config.getOptionalString(KUBERNETES_PLUGIN_KEY);
+  const _logTemplate = `Hub cluster ${OCM_PREFIX}.${id}.${KUBERNETES_PLUGIN_KEY}=${name}`;
+
+  const hub = globalConfig
+    .getConfigArray(KUBERNETES_PLUGIN_CONFIG)
+    .flatMap(method => method.getOptionalConfigArray('clusters') || [])
+    .find(cluster => cluster.getString('name') === name);
+  if (!hub) {
     throw new Error(
-      `'${HUB_CONFIG_PATH}.name' or '${CLUSTER_CONFIG_PATH}.cluster' must be specified in ocm config`,
+      `${_logTemplate} not defined in kubernetes in ${KUBERNETES_PLUGIN_CONFIG}.clusters`,
     );
   }
-  return name;
-};
 
-export const getHubClusterFromKubernetesConfig = (config: Config): Config => {
-  const hub = config
-    .getConfigArray(CLUSTERS_PATH)
-    .flatMap(method => method.getOptionalConfigArray('clusters') || [])
-    .find(
-      cluster =>
-        cluster.getString('name') ===
-        config.getOptionalString(`${CLUSTER_CONFIG_PATH}.cluster`),
-    );
-  if (!hub) {
-    throw new Error('Hub cluster not defined in kubernetes config');
+  if (hub.getString('authProvider') !== 'serviceAccount') {
+    throw new Error(`${_logTemplate} has to authenticate via 'serviceAccount'`);
   }
   return hub;
 };
 
-export const getHubClusterFromOcmConfig = (config: Config): Config => {
-  const hub = config.getConfig(HUB_CONFIG_PATH);
+export const getHubClusterFromOcmConfig = (
+  id: string,
+  config: Config,
+): Config => {
   // Check if required values are valid
   const requiredValues = ['name', 'url'];
```

Comment:
The `serviceAccountToken` has to also be added to the `requiredValues` if it is made required in the config.